android/exponentview/src/main/java/versioned/host/exp/exponent/modules/api/NotificationsModule.java

```diff
@@ -73,8 +96,142 @@ public void onEventFailure(String errorMessage) {
   }
 
   @ReactMethod
+  public void presentLocalNotification(final ReadableMap details, final Promise promise) {
+    ReactApplicationContext context = getReactApplicationContext();
+    final NotificationCompat.Builder builder = new NotificationCompat.Builder(context);
+
+    builder.setSmallIcon(R.drawable.shell_notification_icon);
+    builder.setAutoCancel(true);
+
+    if (!(details.hasKey("silent") && details.getBoolean("silent"))) {
+      builder.setDefaults(NotificationCompat.DEFAULT_SOUND);
+    }
+
+    if (details.hasKey("title")) {
+      String title = details.getString("title");
+      builder.setContentTitle(title);
+      builder.setTicker(title);
+    }
+
+    if (details.hasKey("body")) {
+      builder.setContentText(details.getString("body"));
+    }
+
+    if (details.hasKey("count")) {
+      builder.setNumber(details.getInt("count"));
+    }
+
+    if (details.hasKey("sticky")) {
+      builder.setOngoing(details.getBoolean("sticky"));
+    }
+
+    if (details.hasKey("priority")) {
+      int priority;
+
+      switch (details.getString("priority")) {
+        case "max":
+          priority = NotificationCompat.PRIORITY_MAX;
+          break;
+        case "high":
+          priority = NotificationCompat.PRIORITY_HIGH;
+          break;
+        case "low":
+          priority = NotificationCompat.PRIORITY_LOW;
+          break;
+        case "min":
+          priority = NotificationCompat.PRIORITY_MIN;
+          break;
+        default:
+          priority = NotificationCompat.PRIORITY_DEFAULT;
+      }
+
+      builder.setPriority(priority);
+    }
+
+    if (details.hasKey("vibrate")) {
+      ReadableArray vibrate = details.getArray("vibrate");
+      long[] pattern = new long[vibrate.size()];
+      for (int i = 0; i < vibrate.size(); i++) {
+        pattern[i] = vibrate.getInt(i);
+      }
+      builder.setVibrate(pattern);
+    }
+
+    Activity currentActivity = Exponent.getInstance().getCurrentActivity();
+
+    if (currentActivity == null) {
+      promise.reject(NO_ACTIVITY_ERROR, "Activity not found");
+      return;
+    }
+
+    Intent intent;
+
+    if (details.hasKey("link")) {
+      intent = new Intent(Intent.ACTION_VIEW, Uri.parse(details.getString("link")));
+    } else {
+      Class activityClass = currentActivity.getClass();
+      String manifestUrl = (String) mExperienceProperties.get(KernelConstants.MANIFEST_URL_KEY);
+      intent = new Intent(getReactApplicationContext(), activityClass);
+      intent.putExtra(KernelConstants.MANIFEST_URL_KEY, manifestUrl);
+    }
+
+    PendingIntent contentIntent = PendingIntent.getActivity(context, 0, intent, 0);
+    builder.setContentIntent(contentIntent);
+
+
+    JSONObject notificationPreferences = mManifest.optJSONObject(ExponentManifest.MANIFEST_NOTIFICATION_INFO_KEY);
```

Comment:
we should separate this (color, icon) out into a shared utility that `ExponentGcmListenerService` also uses